### ISSUE TITLE
Zero-dimensional arrays should be accepted as scalars

### ISSUE THREAD

**Issue by burnpanck** [astrofrog/numtraits#14]
Currently, zero-dimensional arrays get rejected on `NumericalTrait(ndim=0)`, as the latter requires the value to be `np.isscalar`. Unfortunately `np.isscalar(np.array(1)) == False`. The error is particularly annoying in conjunction with quantities, as `quantities.Quantity` is a subclass of `np.ndarray` and thus it's instances are not numpy array scalars.

**Comment by astrofrog** [comment]
I agree, it would be nice for this to work. Would you be interested in contributing a patch?

**Comment by burnpanck** [comment]
I have to admit that the validation code looks quite complicated to me (does it have to be that complicated?), so I do not want to mess too much with it. I think I have a fix that works though...
